### DISCUSSION TITLE
What is the recommended method for saving the mesh hierarchy generated by `MeshHierarchy`?

### DISCUSSION THREAD

**Discussion by lrtfm** [firedrakeproject/firedrake#2839] [Firedrake support]
I would like to save the individual meshes produced by MeshHierarchy, but saving them in an `h5` file appears to result in the loss of hierarchy information. Is there a way to restore the hierarchy information of these meshes?

**Comment by dham** [comment]
I don't think so, sorry.

**Reply by wence-** [reply]
You could imagine writing some custom extension that writes the coarse to fine cell maps into the checkpoint file. However, those relationships are locally numbered and specific to the particular parallel distribution of the mesh. If you reload a mesh later there's no guarantee that you'd get the same distribution, and the multigrid mesh hierarchy code relies on all children of a parent cell being on the same process.

That said, if you need this for checkpointing and restart and want to pick up with a new mesh hierarchy afterwards we should probably think of how to enable this.

**Reply by dham** [reply]
I think that's exactly the issue. It just doesn't appear straightforward.

**Reply by lrtfm** [reply]
Thank you for your response. This appears to be complicated.
